packages/docsify-server-renderer/index.js

```diff
@@ -1,18 +1,34 @@
 import { readFileSync } from 'fs';
 import { resolve, basename } from 'path';
-import resolvePathname from 'resolve-pathname';
 import { AbstractHistory } from '../../src/core/router/history/abstract';
 import { Compiler } from '../../src/core/render/compiler';
 import { isAbsolutePath } from '../../src/core/router/util';
 import * as tpl from '../../src/core/render/tpl';
 import { prerenderEmbed } from '../../src/core/render/embed';
+import resolvePathname from 'resolve-pathname';
 import fetch from 'node-fetch';
 import debug from 'debug';
+import DOMPurify from 'dompurify';
 
 function cwd(...args) {
   return resolve(process.cwd(), ...args);
 }
 
+// Borrowed from https://j11y.io/snippets/getting-a-fully-qualified-url.
+function qualifyURL(url){
+    var img = document.createElement('img');
+    img.src = url; // set string url
+    url = img.src; // get qualified url
+    img.src = ''; // prevent the server request
+    return url;
+}
+
+function isExternal(url) {
+  url = qualifyURL(url)
+  url = new URL(url)
+  return url.origin !== location.origin
+}
```

Comment:
Here's the new `isExternal` check, with no room for human error because the URLs are handled the same as the internal browser engine.

GitHub wouldn't let me make a PR to your branch for some reason, so I made a new commit in the same branch.